outport/factory/outportFactory_test.go

```diff
@@ -1,51 +1,124 @@
-package factory
+package factory_test
 
 import (
 	"errors"
 	"testing"
 
+	covalentFactory "github.com/ElrondNetwork/covalent-indexer-go/factory"
 	indexerFactory "github.com/ElrondNetwork/elastic-indexer-go/factory"
 	"github.com/ElrondNetwork/elrond-go/outport"
+	"github.com/ElrondNetwork/elrond-go/outport/factory"
+	"github.com/ElrondNetwork/elrond-go/process/mock"
+	stateMock "github.com/ElrondNetwork/elrond-go/testscommon/state"
 	notifierFactory "github.com/ElrondNetwork/notifier-go/factory"
 	"github.com/stretchr/testify/require"
 )
 
-func createMockArgsOutportHandler() *OutportFactoryArgs {
-	mockElasticArgs := &indexerFactory.ArgsIndexerFactory{}
-	mockNotifierArgs := &notifierFactory.EventNotifierFactoryArgs{}
-	return &OutportFactoryArgs{
-		ElasticIndexerFactoryArgs: mockElasticArgs,
-		EventNotifierFactoryArgs:  mockNotifierArgs,
+func createMockArgsOutportHandler(indexerEnabled, notifierEnabled, covalentEnabled bool) *factory.OutportFactoryArgs {
+	mockElasticArgs := &indexerFactory.ArgsIndexerFactory{
+		Enabled: indexerEnabled,
+	}
+	mockNotifierArgs := &notifierFactory.EventNotifierFactoryArgs{
+		Enabled: notifierEnabled,
+	}
+	mockCovalentArgs := &covalentFactory.ArgsCovalentIndexerFactory{
+		Enabled: covalentEnabled,
+	}
+	return &factory.OutportFactoryArgs{
+		ElasticIndexerFactoryArgs:  mockElasticArgs,
+		EventNotifierFactoryArgs:   mockNotifierArgs,
+		CovalentIndexerFactoryArgs: mockCovalentArgs,
 	}
 }
 
 func TestNewIndexerFactory(t *testing.T) {
 	t.Parallel()
 	tests := []struct {
 		name     string
-		argsFunc func() *OutportFactoryArgs
+		argsFunc func() *factory.OutportFactoryArgs
 		exError  error
 	}{
 		{
 			name: "NilArgsOutportFactory",
-			argsFunc: func() *OutportFactoryArgs {
+			argsFunc: func() *factory.OutportFactoryArgs {
 				return nil
 			},
 			exError: outport.ErrNilArgsOutportFactory,
 		},
 		{
 			name: "AllOkShouldWork",
-			argsFunc: func() *OutportFactoryArgs {
-				return createMockArgsOutportHandler()
+			argsFunc: func() *factory.OutportFactoryArgs {
+				return createMockArgsOutportHandler(false, false, false)
 			},
 			exError: nil,
 		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			_, err := CreateOutport(tt.argsFunc())
+			_, err := factory.CreateOutport(tt.argsFunc())
 			require.True(t, errors.Is(err, tt.exError))
 		})
 	}
 }
+
+func TestCreateOutport_EnabledDrivers_NilMockArgs_ExpectErrorSubscribingDrivers(t *testing.T) {
```

Comment:
TestCreateOutport_EnabledDriversNilMockArgsExpectErrorSubscribingDrivers

Reply:
Renamed it